### ISSUE TITLE
Better error message for generic rest

### ISSUE THREAD

**Issue by rayslxu** [stanford-oval/thingpedia-client#32]
Return better error messages than "Unexpected HTTP error 404".
If there is a more concrete error message in the returned json, return that, otherwise, return an empty list.

**Comment by gcampax** [comment]
What device needs this? Why do we want to swallow a 404? It could hide an actual bug, or prevent us from noticing if the API is taken down.

**Comment by rayslxu** [comment]
There are other error codes also. This is just one example. Anyway, this should not be shown to the users. This should see a more concrete reason such as "user not found" / "credential expired" / etc.

**Comment by gcampax** [comment]
Yeah I'm more worried about returning an empty list silently - in any case, we should do that only for 404, not any other error.
